### PR TITLE
Implement import renaming with `_` (RFC 2166)

### PR DESCRIPTION
**Pull request by petrochenkov**
cc https://github.com/rust-lang/rust/issues/48216

### CODE REVIEW

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -7037,7 +7037,11 @@ impl<'a> Parser<'a> {
 
     fn parse_rename(&mut self) -> PResult<'a, Option<Ident>> {
         if self.eat_keyword(keywords::As) {
-            self.parse_ident().map(Some)
+            if self.eat(&token::Underscore) {
+                Ok(Some(Ident::with_empty_ctxt(Symbol::gensym("_"))))
```
Comment (nikomatsakis) [inline src/libsyntax/parse/parser.rs]: Oh man what a hack. But clever =)

**Review by nikomatsakis — APPROVED** [review]

**Comment by nikomatsakis** [comment]
er, mistake, sorry

**Comment by kennytm** [comment]
@bors r-

The UI tests in this PR will be broken by the rephrasing in #48684. Please rebase and update `src/test/ui/feature-gate-underscore-imports.stderr`. The last line should read

```
For more information about this error, try `rustc --explain E0658`.
```

**Pull request merged**